### DISCUSSION TITLE
PyCharm displays module imports as erroneous even though they work fine

### DISCUSSION THREAD

**Discussion by betterthanever2** [DavidVujic/python-polylith#98] [Q&A] [answered]
Hey all.
I recently encountered an issue with PyCharm: it underlines module imports as though they are wrong, but they still work perfectly fine.
![impr](https://github.com/DavidVujic/python-polylith/assets/68774738/1ece9be9-eb09-4507-9d35-5e6c6372a7cf)

Since everything _is_ working, it's not a big issue, but it's rather irritating.
@DavidVujic , or anybody else - what might be wrong? Reindexing the project didn't help.

**Comment by DavidVujic** [comment, marked as answer]
Hello!

I think it might be an issue with how PyCharm is adding the virtual environment to a repo.

Make sure that you have a local virtual environment configuration, by adding a `poetry.toml` to the top folder with the content:
```toml
[virtualenvs]
path = ".venv"
in-project = true
```

Run `poetry install` in a shell.

This will install the dependencies, and make the environment aware of the bases and components folders. PyCharm will ask about what interpreter to use when opening a Python file. Make sure to choose the local one in the `.venv` folder.

Everything should now be set up correctly for PyCharm!

![pycharm-venv](https://github.com/DavidVujic/python-polylith/assets/301286/e8f235b4-2a65-473b-b570-c97d0dcea5cc)


The[ docs have also been updated ](https://davidvujic.github.io/python-polylith-docs/ide/)with this information. Thank you for posting this here @betterthanever2!

**Reply by betterthanever2** [reply]
It appears the issue was, indeed, with the environment. I went through the steps, and it seems to be fixed now, thank you!
